src/ReefModEngine.jl

```diff
@@ -26,9 +26,11 @@ end
 """
 Only for use when RME functions return non-error numeric results. 
 
-For example any getter retrieving a numberic property.
+# Examples
 
-ivOutplantCountPerM2(::Cstring)::Cdouble.
+```julia
+count::Float64 = @getRME ivOutplantCountPerM2("iv_name"::Cstring)::Cdouble.
+```
```

Comment:
Hey @DanTanAtAims what's the difference between the two below?

```
@getRME ivOutplantCountPerM2("iv_name"::Cstring)::Cdouble

# and

@RME ivOutplantCountPerM2("iv_name"::Cstring)::Cdouble
```

Reply:
The second will error out when there was no error. In the RME macro 
```julia
macro RME(func)
    local m = esc(Meta.parse("@ccall RME.$(func)"))
    return quote
        local err = $m

---->   if !(typeof(err) <: Cstring) && err != 0 && !(typeof(err) <: Cvoid) # <--- this condition prevents function calls that return numbers to fail.
            local err_msg = @ccall RME.lastError()::Cstring
            throw(ArgumentError("Call to RME failed with message: $(unsafe_string(err_msg))"))
        end
        ...
    end
end
```
The conditions of the if statement are evaluated as follows for getters that return numeric, non-error values
```julia
!(Cdouble <: Cstring) (true)
err != 0 (true) # Importantly no error has occurred. 
!(typeof(err) <: Cvoid) (true)
```
So we enter the error `if` block. We should only use the new macro `@getRME` when we have no choice as we lose the safety of the error checking in the original macro.

Reply:
Yep, obvious now, sorry.